_posts/2023-05-02-12.0.0-release.md

```diff
@@ -0,0 +1,265 @@
+* NaNs are correctly handled now for Parquet predicate push-downs ([GH-18481](https://github.com/apache/arrow/issues/18481))
+* Added support for reading Parquet page indexes ([GH-33596](https://github.com/apache/arrow/issues/33596)) and writing page indexes ([GH-34053](https://github.com/apache/arrow/issues/34053))
+* Parquet writer can write columns in parallel now ([GH-33655](https://github.com/apache/arrow/issues/33655))
+* Fixed incorrect number of rows in Parquet V2 page headers ([GH-34086](https://github.com/apache/arrow/issues/34086))
+* Fixed incorrect Parquet page null_count when stats are disabled ([GH-34326](https://github.com/apache/arrow/issues/34326))
+* Added support for reading BloomFilters to the Parquet Reader ([GH-34665](https://github.com/apache/arrow/issues/34665))
+* Parquet File-writer can now add additional key-value metadata after it has been opened ([GH-34888](https://github.com/apache/arrow/issues/34888))
+
+### ORC
+
+* Added support for the union type in ORC writer ([GH-34262](https://github.com/apache/arrow/issues/34262))
+* Fixed ORC CHAR type mapping with Arrow ([GH-34823](https://github.com/apache/arrow/issues/34823))
+* Fixed timestamp type mapping between ORC and arrow ([GH-34590](https://github.com/apache/arrow/issues/34590))
+
+### Datasets
+
+* Added support for reading JSON datasets ([GH-33209](https://github.com/apache/arrow/issues/33209))
+* Dataset writer now supports specifying a function callback to construct the file name in addition to the existing file name template ([GH-34565](https://github.com/apache/arrow/issues/34565))
+
+### Filesystems
+
+* GcsFileSystem::OpenInputFile avoids unnecessary downloads ([GH-34051](https://github.com/apache/arrow/issues/34051))
+
+### Other changes
+
+* Convenience Append(std::optional<T>...) methods have been added to array builders ([GH-14863](https://github.com/apache/arrow/issues/14863))
+* A deprecated OpenTelemetry header was removed from the Flight library ([GH-34417](https://github.com/apache/arrow/issues/34417))
+* Fixed crash in "take" kernels on ExtensionArrays with an underlying dictionary type ([GH-34619](https://github.com/apache/arrow/issues/34619))
+* Fixed bug where the C-Data bridge did not preserve nullability of map values on import ([GH-34983](https://github.com/apache/arrow/issues/34983))
+* Added support for EqualOptions to RecordBatch::Equals ([GH-34968](https://github.com/apache/arrow/issues/34968))
+* zstd dependency upgraded to v1.5.5 ([GH-34899](https://github.com/apache/arrow/issues/34899))
+* Improved handling of "logical" nulls such as with union and RunEndEncoded arrays ([GH-34361](https://github.com/apache/arrow/issues/34361))
+* Fixed incorrect handling of uncompressed body buffers in IPC reader, added IpcWriteOptions::min_space_savings for optional compression optimizations ([GH-15102](https://github.com/apache/arrow/issues/15102))
+
+## C# notes
```

Comment:
```suggestion
## C# notes

* Added support for Float 16 [GH-25163](https://github.com/apache/arrow/issues/25163)
* Added decompression support for Record Batches [GH-32240](https://github.com/apache/arrow/issues/32240) and added new Apache.Arrow.Compression package.
* Define C Data Interface for schemas and types [GH-34737](https://github.com/apache/arrow/issues/34737)
```
@westonpace @wjones127 @eerhardt 
I tried to add some of the relevant notes I found, let me know if those are ok with you?
This is the link with the C# issues closed on the 12.0.0 release

Reply:
I think I added these (and maybe a few more) in https://github.com/raulcd/arrow-site/pull/3 .  Did you want to merge that?